src/flb_pack.c

```diff
@@ -293,3 +293,193 @@ void flb_pack_print(char *data, size_t bytes)
     }
     msgpack_unpacked_destroy(&result);
 }
+
+
+static inline int try_to_write(char *buf, int *off, size_t left,
+                        char *str, size_t str_len)
+{
+    if (str_len <= 0){
+        str_len = strlen(str);
+    }
+    if (left <= *off+str_len) {
+        return FLB_FALSE;
+    }
+    memcpy(buf+*off, str, str_len);
+    *off += str_len;
+    return FLB_TRUE;
+}
+
+static int msgpack2json(char *buf, int *off, size_t left, msgpack_object *o)
+{
+    int ret = FLB_FALSE;
+    int i;
+    int loop;
+
+    switch(o->type) {
+    case MSGPACK_OBJECT_NIL:
+        ret = try_to_write(buf, off, left, "null", 4);
+        break;
+
+    case MSGPACK_OBJECT_BOOLEAN:
+        ret = try_to_write(buf, off, left,
+                           (o->via.boolean ? "true":"false"),0);
+        
+        break;
+
+    case MSGPACK_OBJECT_POSITIVE_INTEGER:
+        {
+            char temp[32] = {0};
+            ret = try_to_write(buf, off, left, temp,
+                 snprintf(temp, sizeof(temp)-1, "%lu", (unsigned long)o->via.u64));
```

Comment:
Could you tell me the reason?  It needs additional variable.
I read CONTRIBUTING.md and Apache Developers' C Language Style Guide,
but I can't find it.

Passing snprintf  works correctly. I wrote a sample code.
https://gist.github.com/nokute78/dfb8063a8d605d5998d767d63b3eb727